replicate-to-sql-with-generic-events/lib/index.js

```diff
@@ -0,0 +1,55 @@
+const { PostgresWrapper, pgConfigDocker } = require('./postgres');
+
+class CorePlugin {
+  constructor() {
+    this.context = null;
+    this.config = {};
+    this.pipes = {
+      'generic:document:afterWrite': 'afterWrite',
+      'generic:document:afterDelete': 'afterDelete'
+    };
+  }
+
+  init(customConfig, context) {
+    this.config = Object.assign(this.config, customConfig);
+    this.context = context;
+    this.pg = new PostgresWrapper(pgConfigDocker);
+    this.pg.connect().then(() => console.log('database connected'));
```

Comment:
The `async` method can return a promise, when it's the case, Kuzzle will wait for this promise fulfillment before starting to process requests.  
So here it would be better to return that promise so Kuzzle will wait for PG client to be connected.  
Also, if the promise is rejected here, it will trigger an unhandledrejection

Reply:
Didn't knew you could return a promise, but your remark lead me to read that part https://docs.kuzzle.io/core/2/plugins/guides/manual-setup/init-function/ which explain it very well :+1: